### PR TITLE
Allow setting custom roles to users on projects

### PR DESCRIPTION
**Pull request by dfredell**
Closes https://github.com/xanzy/go-gitlab/issues/1867 
Prerequisite for https://gitlab.com/gitlab-org/terraform-provider-gitlab/-/issues/6186

Documented API https://docs.gitlab.com/ee/api/members.html#add-a-member-to-a-group-or-project

### CODE REVIEW

**Comment by dfredell** [comment]
I wired it up locally to terraform and confirmed that project and group setting of member_role_id worked.

```tf

locals {
  # Guest role w/ Read code permission
  # https://gitlab.com/groups/flywheel-io/-/settings/roles_and_permissions
  custom_role_readonly_id = 1000640
  # group ID of the readonly team users
  team_readonly_id = 81031499
}


// Create membership entries for each
resource "gitlab_group_membership" "members" {
  # for_each = local.user_group_access_map

  group_id     = local.team_readonly_id
  user_id      = 19712725
  access_level = "guest"
  member_role_id = local.custom_role_readonly_id
}

// Create membership entries for each
resource "gitlab_project_membership" "members" {
  # for_each = local.user_group_access_map

  project     = 14305943
  user_id      = 19712725
  access_level = "guest"
  member_role_id = local.custom_role_readonly_id
}
```

image

On a group:
image

On a project:
image

File: group_members.go [hunk 1]
```diff
@@ -126,9 +126,10 @@ func (s *GroupsService) ListAllGroupMembers(gid interface{}, opt *ListGroupMembe
 // GitLab API docs:
 // https://docs.gitlab.com/ee/api/members.html#add-a-member-to-a-group-or-project
 type AddGroupMemberOptions struct {
-	UserID      *int              `url:"user_id,omitempty" json:"user_id,omitempty"`
-	AccessLevel *AccessLevelValue `url:"access_level,omitempty" json:"access_level,omitempty"`
-	ExpiresAt   *string           `url:"expires_at,omitempty" json:"expires_at"`
+	UserID       *int              `url:"user_id,omitempty" json:"user_id,omitempty"`
+	AccessLevel  *AccessLevelValue `url:"access_level,omitempty" json:"access_level,omitempty"`
+	ExpiresAt    *string           `url:"expires_at,omitempty" json:"expires_at"`
+	MemberRoleId *int              `url:"member_role_id,omitempty" json:"member_role_id,omitempty"`
```
Comment (svanharmelen) [inline group_members.go:132]: In Go this should be called `MemberRoleID`: https://go.dev/wiki/CodeReviewComments#initialisms

**Review by svanharmelen — CHANGES_REQUESTED** [review]
If you can update the name, I'm good with this one...

File: group_members.go (hunk 1 above)
Comment (dfredell) [reply group_members.go]: Update, thanks for taking a look :)

**Review by svanharmelen — APPROVED** [review]
Thanks @dfredell 👍🏻

**Pull request merged**